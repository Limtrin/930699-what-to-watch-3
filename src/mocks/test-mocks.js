export const filmsList = [
  {
    id: `shutter-island`,
    name: `Shutter Island`,
    genre: `detective`,
    year: 2010,
    image: `img/shutter-island.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 22m`,
    rating: 9.5,
    votes: 123,
    director: `Frank Spaskukotski`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      }
    ],
    starring: [
      `Tim Robbins`,
      `Morgan Freeman`,
      `Bob Gunton`,
      `William Sadler`,
      `Clancy Brown`,
      `Gil Bellows`,
      `James Whitmore`
    ],
  },
  {
    id: `snatch`,
    name: `Snatch`,
    genre: `comedy`,
    year: 2009,
    image: `img/snatch.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 22m`,
    rating: 9.5,
    votes: 123,
    director: `Frank Spaskukotski`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      }
    ],
    starring: [
      `Tim Robbins`,
      `Morgan Freeman`,
      `Bob Gunton`,
      `William Sadler`,
      `Clancy Brown`,
      `Gil Bellows`,
      `James Whitmore`
    ],
  }
];

export const film = {
  id: `war-of-the-worlds`,
  name: `War of the Worlds`,
  genre: `action`,
  year: 2008,
  image: `img/war-of-the-worlds.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  cover: `img/bg-the-grand-budapest-hotel.jpg`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  time: `2h 22m`,
  rating: 9.5,
  votes: 123,
  director: `Frank Spaskukotski`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
  reviews: [
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    },
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    },
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    }
  ],
  starring: [
    `Tim Robbins`,
    `Morgan Freeman`,
    `Bob Gunton`,
    `William Sadler`,
    `Clancy Brown`,
    `Gil Bellows`,
    `James Whitmore`
  ],
};
