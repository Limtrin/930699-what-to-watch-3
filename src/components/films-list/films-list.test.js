import React from "react";
import renderer from "react-test-renderer";
import FilmsList from "./films-list.jsx";
import {filmsList} from "../../mocks/test-mocks.js";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space.js";

const mockStore = configureStore([]);

it(`<FilmsList /> should render correctly`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      genre: `All genres`,
      filmsList,
      filmsCurrent: filmsList,
      showedFilms: filmsList
    }
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <FilmsList
            onHeaderClickHandler = {() => {}}
            onFilmCardClickHandler = {() => {}}
            onItemEnterHandler = {() => {}}
            onItemLeaveHandler = {() => {}}
          />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
