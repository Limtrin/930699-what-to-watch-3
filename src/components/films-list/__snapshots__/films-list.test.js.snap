// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmsList /> should render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        onLoadStart={[Function]}
        onPlay={[Function]}
        poster="img/shutter-island.jpg"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/films/shutter-island"
        onClick={[Function]}
      >
        Shutter Island
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        onLoadStart={[Function]}
        onPlay={[Function]}
        poster="img/snatch.jpg"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/films/snatch"
        onClick={[Function]}
      >
        Snatch
      </a>
    </h3>
  </article>
</div>
`;
